### ISSUE TITLE
Support ESCAPE with LIKE

### ISSUE THREAD

**Issue by hlcianfagna** [crate/crate#14790]
### Problem Statement

A specific module of [Ignition](https://inductiveautomation.com/ignition/) runs queries like
```
SELECT ... 
FROM ... 
WHERE ... ***.***** NOT LIKE **** ESCAPE E'\\' ...
```
This currently fails with
```
UnsupportedFeatureException[ESCAPE is not supported.]
```

### Possible Solutions

Support `ESCAPE` with `LIKE` like in https://www.postgresql.org/docs/16/functions-matching.html#FUNCTIONS-LIKE

### Considered Alternatives

In a different context where the query can be modified, the same results can be achieved by tweaking the `LIKE` expression without using `ESCAPE` or using the `~` operator instead.

**Comment by matriv** [comment]
We need to properly implement the E'<escaped_string>' as in postgres:
```
matriv=> select E'Marios\tTrivyzas';
    ?column?      
------------------
Marios  Trivyzas
(1 row)

matriv=> select 'Marios\tTrivyzas';
    ?column?      
------------------
Marios\tTrivyzas
(1 row)
```
and make sure that pattern matching operators (LIKE, ILIKE, regex) behave correctly, e.g.:
```
matriv=> select 'Marios' LIKE E'Ma\%os';
?column?  
----------
t
(1 row)

matriv=> select 'Marios' LIKE 'Ma\%os';
?column?  
----------
f
(1 row)
```

**Comment by hlcianfagna** [comment]
> We need to properly implement the E'<escaped_string>' 

This may already be implemented
https://cratedb.com/docs/crate/reference/en/5.4/sql/general/lexical-structure.html#string-literals-with-c-style-escapes
?

**Comment by matriv** [comment]
true, thx! but there is already something different vs Postgres:
psql:
```
matriv=> select 'Marios' LIKE E'Ma\%os';
?column?  
----------
t
(1 row)

matriv=> select 'Marios' LIKE 'Ma\%os';
?column?  
----------
f
(1 row)
```
cratedb:
```
cr> select 'Marios' LIKE E'Ma\%os';
+-------+
| false |
+-------+
| FALSE |
+-------+
SELECT 1 row in set (0.001 sec)

cr> select 'Marios' LIKE 'Ma\%os';
+-------+
| false |
+-------+
| FALSE |
+-------+
SELECT 1 row in set (0.001 sec)
```

**Comment by BaurzhanSakhariev** [comment]
From https://www.postgresql.org/docs/16/functions-matching.html#FUNCTIONS-LIKE

> The phrases LIKE, ILIKE, NOT LIKE, and NOT ILIKE are generally treated as operators in PostgreSQL syntax; for example they can be used in expression operator ANY (subquery) constructs, although an ESCAPE clause cannot be included there. In some obscure cases it may be necessary to use the underlying operator names instead.

Should we follow this and support EScAPe only for LIKE/NOT LIKE but not for LIKE + ANY?

**Comment by matriv** [comment]
> Should we follow this and support `ESCAPE` only for LIKE/NOT LIKE and not for `LIKE ANY`?

I'm always supporting consistency. In this case, since we're trying to follow PG behavior, so I don't have a clear answer but I tend to do it for all operators.

**Comment by BaurzhanSakhariev** [comment]
>I'm always supporting consistency. In this case, since we're trying to follow PG behavior, so I don't have a clear answer but I tend to do it for all operators.

Makes sense, thanks. Will do it for all operators then unless there is a concrete strong arguments against it. 

My only argument was "be aligned with PG" which is probably not the strongest one

**Comment by BaurzhanSakhariev** [comment]
ESCAPE is not supported in LIKE ANY statements, similar to PG and also according to the SQL standard.
We can add it if there is a concrete use case. 
https://github.com/crate/crate/pull/15166#issuecomment-1849698619
